### PR TITLE
Fix #634

### PR DESCRIPTION
**Pull request by kkoopa**
This ought to be the proper fix for #634.

### CODE REVIEW

File: nan_private.h [hunk 1]
```diff
@@ -24,16 +24,21 @@ HasPrivate(v8::Local<v8::Object> object, v8::Local<v8::String> key) {
...
 GetPrivate(v8::Local<v8::Object> object, v8::Local<v8::String> key) {
-  HandleScope scope;
 #if NODE_MODULE_VERSION >= NODE_6_0_MODULE_VERSION
   v8::Isolate *isolate = v8::Isolate::GetCurrent();
+  v8::EscapableHandleScope scope(isolate);
   v8::Local<v8::Context> context = isolate->GetCurrentContext();
   v8::Local<v8::Private> private_key = v8::Private::ForApi(isolate, key);
-  return object->GetPrivate(context, private_key);
+  v8::MaybeLocal<v8::Value> v = object->GetPrivate(context, private_key);
+  assert(!v.IsEmpty() && "v should not be empty");
```
Comment (kkoopa) [inline nan_private.h]: This line is arguably unnecessary.

**Comment by JoshuaWise** [comment]
bump

**Comment by grinich** [comment]
another +1 to land this

**Comment by stemcc** [comment]
This would be an extremely helpful fix.

File: nan.h [hunk 2]
```diff
@@ -1122,51 +1136,53 @@ widenString(std::vector<uint16_t> *ws, const uint8_t *s, int l) {
   NewOneByteString(const uint8_t * value, int length = -1) {
     std::vector<uint16_t> wideString;  // NOLINT(build/include_what_you_use)
     imp::widenString(&wideString, value, length);
-    return imp::Factory<v8::String>::return_t(v8::String::New(
-        &wideString.front(), static_cast<int>(wideString.size())));
+    return v8::String::New(&wideString.front(),
```
Comment (bnoordhuis) [inline nan.h]: Using .front() when the string is empty is UB.  You could use .data() instead, that has well-defined behavior for the .empty() case (pointer to a `\0` byte as of C++11, something I learned earlier this week.)

**Comment by bnoordhuis** [comment]
Sorry for being so late to reply.  I was on holiday until last week and I had 1,000+ unread emails when I came back.  I will admit to not having read them all.

File: nan.h (hunk 2 above)
Comment (kkoopa) [inline nan.h]: Alright, will do.

**Comment by kkoopa** [comment]
No problem. Thanks for the review. I'll get this sorted out now.

**Comment by kkoopa** [comment]
Done.

**Comment by JoshuaWise** [comment]
Thanks!

**Pull request closed without merging**